pennylane_lightning/src/simulator/StateVectorRaw.hpp

```diff
@@ -74,14 +74,6 @@ class StateVectorRaw
         }
     }
 
-    StateVectorRaw(const StateVectorRaw &) = default;
-    StateVectorRaw(StateVectorRaw &&) noexcept = default;
-
-    auto operator=(const StateVectorRaw &) -> StateVectorRaw & = default;
-    auto operator=(StateVectorRaw &&) noexcept -> StateVectorRaw & = default;
-
-    ~StateVectorRaw() = default;
-
```

Comment:
Curious why we remove the default constructors definitions.

Reply:
I followed the rule of zero https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rc-zero